pkgs/build-support/setup-hooks/patch-ant-buildfiles.sh

```diff
@@ -0,0 +1,30 @@
+
+
+# Sets the `modificationtime` attribute on the <jar> and <war> ant tasks
+# This is done to make the final archives deterministic
+patchAntBuildfile() {
+    local buildfile="$1"
+    # using `local-name()` ignores the `xmlns` attribute in a parent node
+    @xmlstarlet@ ed --inplace \
+        --var T '//*[local-name()="jar" or local-name()="war"]' \
+        --update '$T/@modificationtime' -v 0 \
+        --insert '$T[not(@modificationtime)]' -t attr -n 'modificationtime' -v 0 \
+        "$buildfile"
+}
+
+patchAntBuildfilesHook() {
+    find "$NIX_BUILD_TOP" -name '*.xml' -type f -writable -print0 |
```

Comment:
Can we use something other than $NIX_BUILD_TOP? Rationale: https://github.com/NixOS/nixpkgs/issues/189691.

Reply:
I think it's pretty customary for these hooks to just assume `.` is where things should run. IMO that's even *better* because some builds do `cd some-othe-place` early in the builder and expect the builder to run from that subdirectory.

Reply:
I agree that using `.` is fine, because when `postPatch` runs, the builder will almost always be in `sourceRoot`, the lowest "editable" directory.

And if this misses one very niche case, we can always just update the script or call `patchAntBuildfiles` manually.